### PR TITLE
Added prop blockDocument to block field in user profile page

### PR DESCRIPTION
**Pull request by jailsondamasceno**
#### What did you change? \*

A property has been added to the module called blockDocument

#### Why? \*
The customer needed to block the document field (cpf). Once added, the user should no longer edit. The customer wanted unique accounts per user.

#### How to test it? \*

1. Pass the blockDocument property in the declaration of the my-account app in the store

2. Link the vtex.profile-forme app on the tksio-549 branch in the my-account module. It is a dependency.

3. Link the three store / my-account / profile-form repositories and change the property value in the store. By default the speaker is underfid.
If you do not pass the property or pass the false value, the field allows changes.
If it is set to true and the field still has no saved value, the field will also allow editing, if the field is already saved, the field is blocked.

https://profileform--tokstokio.myvtex.com/account/#/profile/edit

#### prop
image

#### Before
bofore

#### After
affter

#### Related to / Depends on?

https://listastokstok.atlassian.net/browse/TKSIO-549

#### Types of changes \*

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical improvements

### CODE REVIEW

**Comment by jailsondamasceno** [comment]
The files that are giving error in the Lint check have not been changed in this feature, they are already like this. There's something with the addresses. I just messed with my account

File: react/components/pages/ProfileEdit.tsx [hunk 1]
```diff
@@ -38,15 +39,18 @@ interface Props extends InjectedContentWrapperProps {
...
+const enhance = compose<Props, { blockDocument?: boolean}>(
   graphql(GET_PROFILE),
   branch<Props>(
     ({ data }) => data.profile == null,
     renderComponent(ProfileEditLoading)
   ),
-  withProps(({ data }: Props) => ({ profile: data.profile })),
+  withProps(({ data, blockDocument }: Props) => {
+    return { profile: data.profile, blockDocument }
+  }),
```
Comment (salesfelipe) [inline react/components/pages/ProfileEdit.tsx:53]: No need for this change

File: react/index.tsx [hunk 2]
```diff
@@ -5,7 +5,14 @@ import AppRouter from './components/AppRouter'
...
 
-class MyAccount extends Component {
+interface MyAccountProps {
+  blockDocument?: boolean
+}
+
+class MyAccount extends Component<MyAccountProps> {
+  constructor(props: MyAccountProps) {
+    super(props)
+  }
```
Comment (salesfelipe) [inline react/index.tsx:15]: No need for this change

File: react/index.tsx [hunk 3]
```diff
@@ -5,7 +5,14 @@ import AppRouter from './components/AppRouter'
 import Wrapper from './components/MyAccountWrapper'
 import { logGeneralErrors } from './utils/splunk'
 
-class MyAccount extends Component {
+interface MyAccountProps {
```
Comment (salesfelipe) [inline react/index.tsx:8]:
```suggestion
interface Props {
```

File: react/index.tsx [hunk 4]
```diff
@@ -5,7 +5,14 @@ import AppRouter from './components/AppRouter'
...
+  blockDocument?: boolean
+}
+
+class MyAccount extends Component<MyAccountProps> {
```
Comment (salesfelipe) [inline react/index.tsx:12]:
```suggestion
class MyAccount extends Component<Props> {
```

File: react/components/Profile/ProfileFormBox.tsx [hunk 5]
```diff
@@ -131,7 +132,8 @@ interface InnerProps {
 interface OutterProps {
   onDataSave: () => void
   onError: (error: any) => void
-  profile: Profile
+  profile: Profile,
```
Comment (salesfelipe) [inline react/components/Profile/ProfileFormBox.tsx:135]:
```suggestion
  profile: Profile
```

File: react/components/AppRouter.tsx [hunk 6]
```diff
@@ -15,7 +15,15 @@ import AddressCreate from './pages/AddressCreate'
 import AddressEdit from './pages/AddressEdit'
 import Menu from './Menu'
 
-class AppRouter extends Component {
+interface AppRouterProps {
```
Comment (salesfelipe) [inline react/components/AppRouter.tsx:18]:
```suggestion
interface Props {
```

File: react/components/AppRouter.tsx [hunk 7]
```diff
@@ -15,7 +15,15 @@ import AddressCreate from './pages/AddressCreate'
...
+  blockDocument?: boolean
+}
+
+class AppRouter extends Component<AppRouterProps> {
+  constructor(props: AppRouterProps) {
+    super(props)
+  }
```
Comment (salesfelipe) [inline react/components/AppRouter.tsx:25]:
```suggestion
```

File: react/components/AppRouter.tsx [hunk 8]
```diff
@@ -15,7 +15,15 @@ import AddressCreate from './pages/AddressCreate'
 import AddressEdit from './pages/AddressEdit'
 import Menu from './Menu'
 
-class AppRouter extends Component {
+interface AppRouterProps {
+  blockDocument?: boolean
+}
+
+class AppRouter extends Component<AppRouterProps> {
```
Comment (salesfelipe) [inline react/components/AppRouter.tsx:22]:
```suggestion
class AppRouter extends Component<Props> {
```

File: docs/README.md [hunk 9]
```diff
@@ -39,6 +39,74 @@ Now, create the file `store/interfaces.json` and define some interfaces:
...
+```json
+"my-account": {
+    "props": { "blockDocument": true },
+    "blocks": [
+    ]
+  }
+```
+| Prop name              | Type             | Description                                                                                                                                                                                           | Default value     |
+| ---------------------- | ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------- |
+| `blockDocument` | `boolean` | Enables or disables editing the document field in my account  | `underfid` |
```
Comment (salesfelipe) [inline docs/README.md:52]:
```suggestion
| `blockDocument` | `boolean` | Enables or disables editing the document field in MyAccount  | `undefined` |
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added 
+- Added prop `blockDocument` to Enables or disables editing the document field in my account
```
Comment (salesfelipe) [inline CHANGELOG.md:11]: Hi, first of all, thank you for your contribution. Just a couple of changes.
```suggestion
- Prop `blockDocument` to Enables or disables editing the document field in MyAccount
```

**Review by salesfelipe — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 10 above)
Comment (jailsondamasceno) [reply CHANGELOG.md:11]: @salesfelipe  Hey, how's it going? I made all the suggested adjustments! I'm available.

Comment (salesfelipe) [reply CHANGELOG.md:11]: Good, what about yourself? I think you forgot to push the changes... Also, there is a conflict on the changelog

Comment (jailsondamasceno) [reply CHANGELOG.md]: I had commented before the push. Which branch should I update with, the master?

**Comment by jailsondamasceno** [comment]
@salesfelipe They have been corrected.

File: CHANGELOG.md (hunk 10 above)
Comment (jailsondamasceno) [reply CHANGELOG.md]: resolved

**Review by salesfelipe — APPROVED** [review]

**Comment by salesfelipe** [comment]
@jailsondamasceno Thanks again for your contribution 🙏 🥳 🚀

**Pull request merged**